docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,222 @@
+
+For some specific values, there will be the possibility to set them globally, as they will be used in that case for every component created, as defined in the [configuration precedence](#configuration-precedence) section. This values will be documented as being `global` configuration values.
+
+### Developer friendly flags
+
+One of the goals of `odo` is to be easy to use for developers. This forces to translate some configuration possible in the underlying technology (OpenShift/Kubernetes) into a friendlier name that developers could easily understand. An example of this would be the amount of memory a component needs. In OpenShift, this is reflected by limits.memory and requests.memory, and dependening how you set these values influence the QoS of the deployment. In `odo` these should be simplified to `memory`, `min-memory` and `max-memory`. These are terms that any developer will most likely understand without looking at the documentation.
+
+* When `memory` is set, memory requests and limits are the same.
+* `min-memory` should be memory requests.
+* `max-memory` should be memory limits.
+* `memory` is exclusive with `min-memory` or `max-memory`. If set, min-mem and max-mem have the same value as `memory`.
+
+#### Proposed flags
+
+We need to define and delimit what information will be stored in this file. Configuration which can be local and global will be denoted. Otherwise, the configuration will only be local to the application .odo file.
+
+### Deployment decoration
+
+Configuration used to define how the deployment would be created:
+
+* mem/memory (local/global)
+  * min-mem/min-memory (local/global)
+  * max-mem/max-memory (local/global)
+* cpu (local/global)
+  * min-cpu (local/global)
+  * max-cpu (local/global)
+* health (local)
+  * live-health (local)
+  * ready-health (local)
+* build-image (local)
+  * src-path (local)
+  * artifacts-path (local)
+* runtime-image (local)
+  * src-path (local)
+  * deployment-path (local)
+
+### CLI behavior
+
+Configuration used to define how the CLI should behave:
+
+* app-name: (local) Preferred application name if none is defined
+* component-name: (local) Preferred component name if none is defined
+* odoignore: (local/global) list of files/patterns to ignore by default if no .odoignore file is provided, or when it's cerated. **This configuration can be global, as the developer might want to always ignore specific files (e.g. .git, .odo, ...) when pushing.**
+
+#### <a name="odo-file-format"></a> `odo` file format
+
+The format for both files should be human readable. Whether it's yaml, json or [toml](https://github.com/toml-lang/toml) is not the scope of this proposal to define, as it will be dictated by the engineering team implementing this feature. In any case, it should be possible to:
+
+* Visually interpret the content of a file (vi, cat)
```

Comment:
I assume this 

> Visually interpret the content of a file (vi, cat)
> Process the file in an easy way (grep, aqk, sed, jq|yq)

is only for .odoignore file right? Because to me it( the above 2 points ) seems to suggest that the files will be exposed to manual edits...

Reply:
for all config files. Yes, it should be exposed to manual edits. But it will be suggested not to do that, and use  `odo utils config` commands to manipulate files.

Reply:
Fixed